src/Scripts/Ecs/Systems/CombatSystem.gd

```diff
@@ -0,0 +1,31 @@
+extends "res://Scripts/Ecs/Core/System.gd"
+
+var _event_bus
+
+func _init(event_bus):
+	self.required_component_types = ["HealthComponent"]
+	event_bus.connect('move_entity', self, 'on_move_entity')
+	_event_bus = event_bus
+
+func on_move_entity(attacker, x, y):
```

Comment:
I like this design a lot better than "attack intent entities" described in my comments above. :100: